constraint/pkg/client/drivers/local/local.go

```diff
@@ -664,6 +714,61 @@ func validateTargets(templ *templates.ConstraintTemplate) error {
 	}
 }
 
+// createConstraintGKPath returns the subpath for given a constraint GK.
+func createConstraintGKSubPath(gk schema.GroupKind) string {
+	return "/" + path.Join("cluster", gk.Group, gk.Kind)
+}
+
+// createConstraintSubPath returns the key where we will store the constraint
+// for each target: cluster.<group>.<kind>.<name>.
+func createConstraintSubPath(constraint *unstructured.Unstructured) (string, error) {
+	if constraint.GetName() == "" {
+		return "", fmt.Errorf("%w: missing name", ErrInvalidConstraint)
+	}
+
+	gvk := constraint.GroupVersionKind()
+	if gvk.Group == "" {
```

Comment:
We should check that `gvk.Group == constraints.Group`, since that is the only valid value.

Reply:
(Also add a test for when gvk.Group is set, but to an incorrect value)